### DISCUSSION TITLE
Hot to implement switch color in next.js 14

### DISCUSSION THREAD

**Discussion by bogdan-marian** [chakra-ui/chakra-ui#8113] [Q&A] [answered]
This is my page.js

```javascript
import Image from "next/image";

import {
  Button,
  Flex,
  Heading,
  Input,
  ColorModeScript,
  useColorMode,
  Text,
} from "@chakra-ui/react";
import theme from "./theme";

// this line brakes the app
const { colorMode, toggleColorMode } = useColorMode()

// this is the top page
export default function Page() {
  return (
    <div>
      <ColorModeScript initialColorMode={theme.config.initialColorMode} />
      <Text>Hello chakra 2.2.0 and next.js 14.0.2</Text>
    </div>
  );
}

```

If I add this 
`const { colorMode, toggleColorMode } = useColorMode()`
in front of export 

I get `Error: (0 , _chakra_ui_react__WEBPACK_IMPORTED_MODULE_3__.useColorMode) is not a function`

**Comment by bogdan-marian** [comment, marked as answer]
I found the problem. (2 things I had to change)
- I need to add "use client"; at the top
- mov const definition inside Page fuction before return

like this
```javascript
"use client";
import Image from "next/image";

import {
  Button,
  Flex,
  Heading,
  Input,
  ColorModeScript,
  useColorMode,
  Text,
} from "@chakra-ui/react";
import theme from "./theme";



// this is the top page
export default function Page() {

// this work now
const { colorMode, toggleColorMode } = useColorMode()

  return (
    <div>
      <ColorModeScript initialColorMode={theme.config.initialColorMode} />
      <Text>Hello chakra 2.2.0 and next.js 14.0.2</Text>
    </div>
  );
}
```

**Comment by bogdan-marian** [comment]
Now I have a question. 
Since I want to add some core layout to my entire app does this mean that my app will allways be render frontned only? 
I do not have the benefit of rendering things on server side if I use next.js?

**Reply by Omikorin** [reply]
Hi! Chakra UI is an Emotion-based framework (at this moment) which is a **runtime** CSS-in-JS.

Please refer to https://chakra-ui.com/getting-started/nextjs-guide#app-directory-setup

**Comment by angelhtml** [comment]
just use "use client"
